docker_push_latest_if_changed.py

```diff
@@ -1,5 +1,182 @@
 #!/usr/bin/env python3
 import argparse
+import hashlib
+import logging
```

Comment:
logging is often hard to trace through with pytest and won't always produce useful output -- probably stick to `print(...)` and/or `print(..., file=...)`

It's also an easy pitfall to fall into instead of using a debugger (use a debugger!)

Reply:
Switched to just using print with a class that handles dealing with the verbosity stuff

I agree a debugger is the best when developing, but I'd like to keep the verbose prints since it's nice to have in logs for CIs like Jenkins.